express-notes-api/server.js

```diff
@@ -0,0 +1,127 @@
+import express from 'express';
+import { readFileSync } from 'node:fs';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+const data = readFileSync('data.json', 'utf-8');
+const dataParse = JSON.parse(data);
+const errorCodeNegative = { error: 'id must be a positive integer' };
+const errorMessage = { error: 'An unexpected error occured' };
+app.use(express.json());
+
+// GET STARTS BELOW
+
+readFile('data.json')
+  .then((success) => {
+    app.get('/api/notes', (req, res) => {
+      const allNotes = [];
+      for (const [key] of Object.entries(dataParse.notes)) {
+        allNotes.push(dataParse.notes[key]);
+      }
+      res.status(200).json(allNotes);
+    });
+  })
+  .catch((error) => {
+    console.log('this is an error message for function read()', error.message);
+    process.exit(1);
+  });
+
+app.get('/api/notes/:id', (req, res) => {
+  const numberId = Number(req.params.id);
+  if (numberId <= 0) {
+    res.status(400).json(errorCodeNegative);
+  } else if (dataParse.notes[numberId]) {
+    res.status(200).json(dataParse.notes[numberId]);
+  } else {
+    const errorMessageNoId = { error: `Cannot find note with id ${numberId}` };
+    res.status(404).json(errorMessageNoId);
+  }
+});
+
+// GET ENDS HERE
+
+// POST STARTS BELOW
+app.post('/api/notes', (req, res) => {
+  const newEntry = req.body;
+  if (newEntry.content === undefined) {
+    const errorMessage = { error: 'Content is a required field' };
+    res.status(404).json(errorMessage);
+  } else if (newEntry.content !== undefined) {
+    newEntry.id = dataParse.nextId;
+    dataParse.notes[dataParse.nextId] = newEntry;
+    dataParse.nextId++;
+    console.log(dataParse);
+    res.status(201).json(newEntry);
```

Comment:
You should probably wait until the write is successful to return a response. Otherwise you won't be able to return an error response if the write fails.